openwisp_controller/config/who_is/service.py

```diff
@@ -107,43 +110,51 @@ def notify_on_failure_static(kwargs, exc):
     @staticmethod
     def is_valid_public_ip_address(ip):
         """
-        Returns True if the given IP address is a valid public IP address.
+        Check if given IP address is a valid public IP address.
         """
         try:
             return ip and ip_address(ip).is_global
         except ValueError:
             return False
 
-    @cached_property
-    def is_whois_enabled(self):
+    @property
+    def is_who_is_enabled(self):
         """
-        Returns True if WhoIs is enabled for the organization of the device.
+        Check if WhoIs is enabled for the organization of the device.
+        The OrganizationConfigSettings are cached as these settings
+        are not expected to change frequently. The timeout for the cache
+        is set to the same as the checksum cache timeout for consistency
+        with DeviceChecksumView.
         """
-        try:
-            device_pk = self.device.pk
-            # here we are fetching the organization settings from db instead of through
-            # device as in some views like `DeviceChecksumView` the device instance is
-            # cached, leading to stale data. Have used Subquery to fetch the settings
-            # in a single query.
-            org_settings = OrganizationConfigSettings.objects.get(
-                organization=Subquery(
-                    Device.objects.filter(pk=device_pk).values("organization_id")[:1]
+        org_pk = self.device.organization.pk
+        org_settings = cache.get(self.ORG_SETTINGS_CACHE_KEY.format(org_pk=org_pk))
```

Comment:
For keeping it DRY (Don't Repeat Yourself), let's create a static method `get_cache_key(org_id)` which would return the cache key for the organization.